modules/imgproc/src/morph.simd.hpp

```diff
@@ -106,12 +106,14 @@ struct MorphNoVec
     int operator()(uchar**, int, uchar*, int) const { return 0; }
 };
 
-#if CV_SIMD
+// Only for clang temporarily. PR #24166
+// TODO: Remove __clang__ macro when GCC is available.
+#if (CV_SIMD || CV_SIMD_SCALABLE) && defined(__clang__)
```

Comment:
Shouldn't it be:
```suggestion
#if (CV_SIMD || (CV_SIMD_SCALABLE && defined(__clang__))
```
Otherwise regular CV_SIMD will not work with GCC and MSVC on other platforms.

Reply:
Both variants don't look correct.

It is not a problem of SCALABLE.
It is related to GCC for RISC-V platform.

BTW, I still recommend to avoid trying to disable code for experimental platforms on mainline branches. Compiler could be fixed, but disabled code is in repository forever.

Reply:
I got your point about the macro that we need to ensure that this code block is only disable on GCC on the RISC-V platform, then we can use `#if CV_SIMD || (CV_SIMD_SCALABLE && !(defined(__gnu_linux__) && defined(__riscv)))` to disable the code block on riscv gcc.

I also got the point that we can keep `#if (CV_SIMD || CV_SIMD_SCALABLE)` and leave the broken part on riscv GCC and wait for GCC upstream to fix it. It also make sence.

Which one do we prefer?

Reply:
Let's disable problematic blocks in SCALABLE mode to avoid breaking more tests, it will help us to establish CI testing. And then we will gradually resolve these issues (whether they are in GCC or qemu or our implementation) and re-enable these blocks. Our main goal is to get rid of obsolete constructions while keeping builds green (gcc-13 and clang-16).

It is not necessary to distinguish between clang and gcc, just leave `#if CV_SIMD` and add a `TODO:` comment.

Reply:
Done
```
#if CV_SIMD // TODO: enable for CV_SIMD_SCALABLE, GCC 13 related
```